filebeat/docs/reference/configuration/filebeat-options.asciidoc

```diff
@@ -383,6 +383,8 @@ lines are discarded. The default is 500.
 
 If this option is set to true, Filebeat starts reading new files at the end of each file instead of the beginning. When this option is used in combination with log rotation, it's possible that the first log entries in a new file might be skipped. The default setting is false.
 
+In case filebeat was run previously and a state was already persisted for the file, `tail_files` will not apply. Harvesting will continue at the previous offset. `tail_files` only applies to files not seen before. To apply `tail_files` to all files, filebeat has to be stopped and the `registry_file` has to be removed. Be aware that this removes ALL previous states.
```

Comment:
Suggested edits:

This setting applies to files that Filebeat has not already processed. If you ran Filebeat previously and the state of the file was already persisted, `tail_files` will not apply. Harvesting will continue at the previous offset. To apply `tail_files` to all files, you must stop Filebeat and remove the `registry_file`. Be aware that doing this removes ALL previous states.